src/frontend/src/optimizer/mod.rs

```diff
@@ -514,6 +514,16 @@ impl PlanRoot {
             ))?;
         }
 
+        if ctx.session_ctx().config().streaming_unaligned_join() {
+            // TODO: make it a logical optimization.
+            // Rewrite joins with index to delta join
```

Comment:
We can check session `streaming_enable_delta_join` to throw an error if both delta join and unaligned have been set to true.